src/crypto/slow-hash.c

```diff
@@ -908,7 +908,13 @@ STATIC INLINE void aes_pseudo_round_xor(const uint8_t *in, uint8_t *out, const u
 void cn_slow_hash(const void *data, size_t length, char *hash, int variant, int prehashed)
 {
     RDATA_ALIGN16 uint8_t expandedKey[240];
+
+#ifndef FORCE_USE_HEAP
     RDATA_ALIGN16 uint8_t hp_state[MEMORY];
+#else
+    RDATA_ALIGN16 uint8_t *hp_state = NULL;
```

Comment:
AFAIKCT, this aligns the pointer, but not the malloc. Typical alignment is 8. I do have a  portable aligned malloc thing which I've not yet PR'd though if that helps.

Reply:
you are right - this aligns the pointer, which is silly. AFAIK ```malloc() ``` aligns correctly anyway, so there is no need for explicit alignment here.

Reply:
For your libc maybe, but I'm used to malloced memory being 8 aligned.

Reply:
no, you are right. I should use
```C
uint8_t *hp_state = (uint8_t *)memalign(16, MEMORY);
```
instead, although the NEON instruction set doesn't seem to care too much about alignment.

Reply:
on second thought, using memalign is a bad idea. @moneromooo-monero you mentioned you have a portable aligned malloc thing ...